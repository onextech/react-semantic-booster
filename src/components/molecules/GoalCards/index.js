import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Card, Icon } from 'semantic-ui-react';
import { MediaCss } from '../../../utils/responsive';


const transitionOptions = '.3s ease-out';
const CardGrid = styled(Card.Group)`
  &.ui.cards {
    .ui.card {
      padding: 1.5em !important;
      .content .header {
        font-size: 2.2em;
        line-height: 1.2;
        text-align: left;
        font-weight: 400;
      }
    }
  }
  
  // hover effects
  &.ui.cards .ui.card {
    .list-links {
      list-style: none;
      text-align: left;
      padding: 0;
      
      .list-link {
        display: inline-block;
        font-size: 1.1em;
        padding: 0.5em 0;
        
        .list-link-icon {
          margin-left: 0.4em;
          transition: transform ${transitionOptions};
        }
        
        &:hover {
          text-decoration: underline;
          &.icon {
            text-decoration: none;
          }
          .list-link-icon {
            transform: translate3D(.7em,0,0);
            transition: transform ${transitionOptions};
          }
        }
      }
    }
    
    // effects
    ${MediaCss.min.sm`
      .content {
        min-height: 18em;
        position: relative;
        overflow: hidden;
        .list-links {
          position: absolute;
          transform: translate3D(0,20em,0);
          transition: transform ${transitionOptions};
          margin-top: 2em;
        }
        .header {
          transform: translate3D(0,3em,0);
          transition: transform ${transitionOptions};
        }
      }
      &:hover {
        .header {
          transform: translate3D(0,0,0);
          transition: transform ${transitionOptions};
        }
        .list-links {
          transform: translate3D(0,0,0);
          transition: transform ${transitionOptions};
        }
      }
    `}
    
  }
`;

const GoalCards = ({ cards }) => (
  <CardGrid itemsPerRow={3} doubling stackable>
    {
      cards.items.map(card => (
        <Card key={card.header}>
          <Card.Content>
            <Card.Header>{card.header}</Card.Header>
            {
              card.listlinks &&
              <ul className="list-links">
                {
                  card.listlinks.map(listlink => (
                    <li className="list-item" key={listlink.name}>
                      <Link className={`list-link ${cards.icon && 'icon'}`} to={listlink.to}>
                        {listlink.name}
                        {cards.icon && <Icon className="list-link-icon" name={cards.icon.name || 'long right arrow'} />}
                      </Link>
                    </li>
                  ))
                }
              </ul>
            }
          </Card.Content>
        </Card>
      ))
    }
  </CardGrid>
);

GoalCards.propTypes = {
  cards: PropTypes.shape({
    items: PropTypes.arrayOf(PropTypes.shape({
      header: PropTypes.string,
      listlinks: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        to: PropTypes.string,
      })),
    })),
    icon: PropTypes.bool,
  }).isRequired,
};

export default GoalCards;
