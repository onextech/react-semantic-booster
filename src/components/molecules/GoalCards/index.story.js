import React from 'react';
import { storiesOf } from '@storybook/react';
import GoalCards from '.';


export const demoCards = { // eslint-disable-line import/prefer-default-export
  items: [
    {
      header: 'Disrupt with game-changing strategies',
      listlinks: [
        { name: 'Data & Analytics', to: '/home' },
        { name: 'Digital Strategy', to: '/about' },
        { name: 'Internet of Things', to: '/work' },
        { name: 'Robotics & Automation', to: '/contact' },
      ],
    },
    {
      header: 'Build a digital culture',
      listlinks: [
        { name: 'Digital Culture & Capabilities', to: '/home' },
        { name: 'Digital Organization', to: '/about' },
        { name: 'Full-Scale Transformation', to: '/work' },
      ],
    },
    {
      header: 'Modernize core technology',
      listlinks: [
        { name: 'Cybersecurity', to: '/home' },
        { name: 'Application Management', to: '/about' },
        { name: 'Enterprise Architecture', to: '/work' },
        { name: 'Infrastructure & Cloud', to: '/work' },
        { name: 'IT Sourcing & Service Providers', to: '/work' },
      ],
    },
    {
      header: 'Create stunning digital experiences',
      listlinks: [
        { name: 'End-To-End Journey Redesign', to: '/home' },
        { name: 'Experience Design', to: '/about' },
      ],
    },
    {
      header: 'Turbo-charge marketing & sales',
      listlinks: [
        { name: 'Digital Marketing Operations', to: '/home' },
        { name: 'Digital Sales', to: '/about' },
        { name: 'Dynamic eCommerce Pricing', to: '/work' },
        { name: 'Personalization at Scale', to: '/work' },
      ],
    },
    {
      header: 'Optimize your operations',
      listlinks: [
        { name: 'Corporate Functions', to: '/home' },
        { name: 'Enterprise Resource Planning', to: '/about' },
        { name: 'IT Project Optimization', to: '/work' },
        { name: 'Lean IT', to: '/work' },
        { name: 'Service Operations', to: '/work' },
      ],
    },
  ],
  icon: true,
};

storiesOf('GoalCards', module)
  .add('Default', () => (
    <GoalCards cards={demoCards} />
  ));
