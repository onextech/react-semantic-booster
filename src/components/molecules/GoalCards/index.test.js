import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import { difference } from 'lodash';
import GoalCards from '.';
import { demoCards } from './index.story';


describe('<GoalCards>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Router><GoalCards cards={demoCards} /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has correct parent classes', () => {
    const wrapper = render(<Router><GoalCards cards={demoCards} /></Router>);
    const renderedClasses = wrapper.attr('class').split(' ');
    const shouldHaveClasses = 'ui doubling stackable cards'.split(' ');
    const areClassesCorrect = difference(shouldHaveClasses, renderedClasses).length === 0;
    expect(areClassesCorrect).toBe(true);
    expect(wrapper.length).toBe(1);
  });
});
