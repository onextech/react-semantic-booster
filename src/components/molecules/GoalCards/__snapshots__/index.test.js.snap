// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoalCards> renders correctly 1`] = `
<div
  className="ui doubling stackable three cards sc-bdVaJa dxmbKJ"
>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Disrupt with game-changing strategies
      </div>
      <ul
        className="list-links"
      >
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/home"
            onClick={[Function]}
          >
            Data & Analytics
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/about"
            onClick={[Function]}
          >
            Digital Strategy
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Internet of Things
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/contact"
            onClick={[Function]}
          >
            Robotics & Automation
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Build a digital culture
      </div>
      <ul
        className="list-links"
      >
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/home"
            onClick={[Function]}
          >
            Digital Culture & Capabilities
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/about"
            onClick={[Function]}
          >
            Digital Organization
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Full-Scale Transformation
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Modernize core technology
      </div>
      <ul
        className="list-links"
      >
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/home"
            onClick={[Function]}
          >
            Cybersecurity
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/about"
            onClick={[Function]}
          >
            Application Management
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Enterprise Architecture
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Infrastructure & Cloud
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            IT Sourcing & Service Providers
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Create stunning digital experiences
      </div>
      <ul
        className="list-links"
      >
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/home"
            onClick={[Function]}
          >
            End-To-End Journey Redesign
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/about"
            onClick={[Function]}
          >
            Experience Design
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Turbo-charge marketing & sales
      </div>
      <ul
        className="list-links"
      >
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/home"
            onClick={[Function]}
          >
            Digital Marketing Operations
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/about"
            onClick={[Function]}
          >
            Digital Sales
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Dynamic eCommerce Pricing
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Personalization at Scale
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Optimize your operations
      </div>
      <ul
        className="list-links"
      >
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/home"
            onClick={[Function]}
          >
            Corporate Functions
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/about"
            onClick={[Function]}
          >
            Enterprise Resource Planning
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            IT Project Optimization
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Lean IT
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            className="list-link icon"
            href="/work"
            onClick={[Function]}
          >
            Service Operations
            <i
              aria-hidden="true"
              className="long right arrow icon list-link-icon"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
